broker/index.js

```diff
@@ -34,323 +36,157 @@ var debug = function() {
+    if (authorized) {
+
+        // check if sensor exists in db
+        database.Sensors.get(client.id)
+        .then(function(sensor) {
+            // if sensor was never registered, create it
+            if(!sensor) {
+                console.log("Creating the sensor")
+                return database.Sensors.create({
+                    'name': pokemon.random(),
+                    'sim': client.id,
+                    'period': CONF.period,
+                    'start_hour': CONF.start_hour,
+                    'stop_hour': CONF.stop_hour
                 })
-                .then(function(sensor){
-                    console.log('sending config');
-                    var date = new Date();
-                    // Send config to the sensor
-                    sendCommand(tcpSocketSensor, 'init '+ [sensor.data_period, sensor.start_time, sensor.stop_time, date.toISOString()].join(" "));
+                .then(function(createdSensor){
+                    return createdSensor;
                 })
                 .catch(function(err) {
-                    console.log("[ERROR] Couldn't get sensor's config in DB :", err);
+                    console.log("[ERROR] Couldn't create sensor :", err);
                 })
-            } else {
-                console.log("Wrong token for authenticating the sensor.");
             }
-        }
-
-        // handle data
-        else if (sim) {
-            handleData(message, sim2socket[sim], sim);
-        }
-
-    });
-
-    tcpSocketSensor.on('close', function() {
-        console.log("connection closed");
-        if (sim) {
-            debug("Removing from sim2socket");
-            delete sim2socket[sim];
-        }
-    });
-
-    tcpSocketSensor.on('error', function(err) {
-        console.log("[ERROR] " + (err ? err.code : "???") + " : " + (err ? err : "unknown"));
-        if (sim) {
-            debug("Removing from sim2socket");
-            delete sim2socket[sim];
-        }
-    });
-
-});
-
-// tcpServerForSensors.on("listening", function(){
-//     // if dev mode simulate data
-//     if (process.env.NODE_ENV === "development") simulateSensorStatusArrivalTCP();
-// })
-
-tcpServerForSensors.on('error', function(err) {
-    console.log("[ERROR] : ", err.message);
-    if (err.code.toString() === 'EADDRINUSE') {
-        console.log("address in use, please retry later ...");
-        process.exit(1);
+            else { 
+                return sensor;
+            }
+            
+        })
+        .then(function(sensor){
+            id2sensor[client.id] = sensor;
+        })
+        .catch(function(err) {
+            console.log("[ERROR] Couldn't get sensor's config in DB :", err);
+        })
     }
-});
 
-tcpServerForSensors.listen(monitorPort);
-
-
-
-
-
-
-
-
-
-// ############### API communication ######################
-
-var tcpServerToAdminApp = net.createServer(function(tcpSocketAdminApp) {
+    callback(null, authorized);
```

Comment:
There was the mother tickling problem :sob: ...
`callback` should be called after the resolution of the promise.